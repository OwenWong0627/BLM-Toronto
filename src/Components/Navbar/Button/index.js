import React from 'react';
import './index.css';
import { BrowserRouter as Router, Link } from 'react-router-dom';

const STYLES = ['btn--primary', 'btn--outline'];

const SIZES = ['btn--medium', 'btn--large'];

export const Button = ({
    children,
    type,
    onClick,
    buttonStyle,
    buttonSize
}) => {
    const checkButtonStyle = STYLES.includes(buttonStyle) ? buttonStyle : STYLES[0];

    const checkButtonSize = SIZES.includes(buttonSize) ? buttonSize : SIZES[0];

    return (
        <Router>
            <Link to='/location-search' className='btn-mobile'>
                <button
                    className={`btn ${checkButtonStyle} ${checkButtonSize}`}
                    onClick={onClick}
                    type={type}
                >
                    {children}
                </button>
            </Link>
        </Router>
    )

};

export default Button;
