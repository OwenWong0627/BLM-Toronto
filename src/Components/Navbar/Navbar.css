.navbar {
  background: linear-gradient(90deg, rgb(237, 238, 238) 0%, rgb(239, 242, 242) 100%);
  height: 80px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  font-size: 1.2rem;
  position: sticky;
  top: 0;
  z-index: 999;
}

.navbar-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 80px;
  max-width: 1500px;
}

.navbar-logo {
  color: #000;
  justify-self: start;
  cursor: pointer;
  text-decoration: none;
  font-size: 2rem;
  display: flex;
  align-items: center;
}

.navbar-logo-image {
  margin-left: 0.5rem;
  width: 1.8rem;
}

.nav-menu {
  display: grid;
  grid-template-columns: repeat(4, auto);
  grid-gap: 10px;
  list-style: none;
  text-align: center;
  width: 50vw;
  justify-content: left;
  margin-left: 2rem;
}

.nav-item {
  height: 80px;
}

.nav-links {
  color: #000;
  display: flex;
  align-items: center;
  text-decoration: none;
  padding: 0.5rem 1rem;
  height: 100%;
}

.nav-links:hover {
  border-bottom: 4px solid #00BFFF;
  transition: all 0.2s ease-out;
}

.menu-icon.fa-bars {
  color: #000;
}

.nav-links-mobile {
  display: none;
}

.menu-icon {
  display: none;
}

@media screen and (max-width: 960px) {
  .NavbarItems {
    position: relative;
  }

  .nav-menu {
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 90vh;
    position: absolute;
    top: 80px;
    left: -100%;
    opacity: 1;
    transition: all 0.5s ease;
    margin-left:0;
  }

  .nav-menu.active {
    background: #DBDEDE;
    left: 0;
    opacity: 1;
    transition: all 0.5s ease;
    z-index: 1;
  }

  .nav-links {
    text-align: center;
    padding: 2rem;
    width: 100%;
    display: table;
  }

  .nav-links:hover {
    background-color: #fff;
    color: #000;
    border-bottom: 4px solid #00BFFF;
    border-radius: 0;
  }

  .menu-icon {
    display: block;
    position: absolute;
    top: 0;
    right: 0;
    transform: translate(-100%, 60%);
    font-size: 1.8rem;
    cursor: pointer;
  }

  .menu-icon.fa-times {
    color: #000;
    font-size: 2rem;
  }

  .nav-links-mobile {
    display: block;
    text-align: center;
    margin: 2rem auto;
    border-radius: 4px;
    width: 80%;
    text-decoration: none;
    font-size: 1.5rem;
    background-color: #00BFFF;
    color: #fff;
    padding: 14px 20px;
    border: 1px solid #fff;
    transition: all 0.3s ease-out;
  }

  .nav-links-mobile:hover {
    background: #50C878;
    color: #000;
    transition: 250ms;
  }
}