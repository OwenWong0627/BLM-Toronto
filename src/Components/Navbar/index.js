import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import './index.css';
import { Button } from './Button';

function Navbar() {
    //initial state of this const is fas fa-bars
    const [click, setClick] = useState(false);
    const [button, setButton] = useState(true);

    //this will handle the switch between fas fa-times and fas fa-bars
    const handleClick = () => setClick(!click);
    const closeMobileMenu = () => setClick(false);

    const showButton = () => {
        if (window.innerWidth <= 960) {
            setButton(false);
        } else {
            setButton(true);
        }
    }

    useEffect(() => {
        showButton();
    }, []);

    window.addEventListener('resize', showButton);
    return (
        <>
            <nav className="navbar">
                <div className="navbar-container">
                    {/*title + logo in the navbar*/}
                    <Link to="/" className="navbar-logo" onClick={closeMobileMenu}>
                        BLM-Ontario <i className='fas fa-thumbs-up' />
                    </Link>
                    <div className='menu-icon' onClick={handleClick}>
                        <i className={click ? 'fas fa-times' : 'fas fa-bars '} />
                    </div>
                    <ul className={click ? 'nav-menu active' : 'nav-menu'}>
                        <li className='nav-item'>
                            <Link
                                to="/"
                                className='nav-links'
                                onClick={closeMobileMenu}
                            >
                                Home
                            </Link>
                        </li>
                        <li className='nav-item'>
                            <Link
                                to="/add-business"
                                className='nav-links'
                                onClick={closeMobileMenu}
                            >
                                Add a Business
                            </Link>
                        </li>
                        <li>
                            <Link
                                to='/location-serach'
                                className='nav-links-mobile'
                                onClick={closeMobileMenu}
                            >
                                Find a Business
                            </Link>
                        </li>
                    </ul>
                    {button && <Button buttonStyle='btn--outline'>Find a Business</Button>}
                </div>
            </nav>
        </>
    )
}

export default Navbar
