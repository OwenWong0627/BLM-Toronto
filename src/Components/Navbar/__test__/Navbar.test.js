import React from 'react';
import ReactDOM from 'react-dom';
import Navbar from '..';

import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it('Renders without crashing', () => {
   const div = document.createElement("div");
   ReactDOM.render(<Router><Navbar /></Router>, div);
});

it('Renders navbar logo correctly and it links to the home page', () => {
   const { getByTestId } = render(<Router><Navbar /></Router>);
   expect(getByTestId('navbar-logo')).toHaveTextContent("BLM-Toronto");
   expect(getByTestId('navbar-logo')).toHaveAttribute("href", "/");
});

const links = [
   { text: 'Home', location: "/" },
   { text: 'Virtual Businesses Near Me', location: "/list-business/" }
];

test.each(links)(
   "Check if Nav Bar has %s link.",
   (link) => {
      render(<Router><Navbar /></Router>);
      const linkDom = screen.getByText(link.text);
      expect(linkDom).toHaveAttribute("href", link.location);
   }
);

test('the button triggers the handleClick and closeMobileMenu function as intended', () => {
   const { getByTestId } = render(<Router><Navbar /></Router>);
   expect(getByTestId('mobile-menu-icon')).toHaveClass('fas fa-bars');
   fireEvent.click(getByTestId('mobile-menu-icon'));
   expect(getByTestId('mobile-menu-icon')).toHaveClass('fas fa-times');
   fireEvent.click(getByTestId('navbar-logo'));
   expect(getByTestId('mobile-menu-icon')).toHaveClass('fas fa-bars');
});

it('Renders the mobile navbar button correctly and it links to the correct page', () => {
   const { getByTestId } = render(<Router><Navbar /></Router>);
   expect(getByTestId('mobile-button')).toHaveTextContent("Find a Business");
   expect(getByTestId('mobile-button')).toHaveAttribute("href", "/find-business");
});

it('matches snapshot', () => {
   const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
   expect(tree).toMatchSnapshot();
});