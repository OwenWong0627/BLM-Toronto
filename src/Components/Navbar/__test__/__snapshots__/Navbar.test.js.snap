// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar"
>
  <div
    className="navbar-container"
  >
    <a
      className="navbar-logo"
      data-testid="navbar-logo"
      href="/"
      onClick={[Function]}
    >
      BLM-Toronto 
      <img
        alt="Navbar Logo"
        className="navbar-logo-image"
        src="/BLM-Toronto-Logo.svg"
      />
    </a>
    <div
      className="menu-icon"
      onClick={[Function]}
    >
      <i
        className="fas fa-bars "
        data-testid="mobile-menu-icon"
      />
    </div>
    <ul
      className="nav-menu"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-links"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-links"
          href="/list-business/"
          onClick={[Function]}
        >
          Virtual Businesses Near Me
        </a>
      </li>
      <li>
        <a
          className="nav-links-mobile"
          data-testid="mobile-button"
          href="/find-business"
          onClick={[Function]}
        >
          Find a Business
        </a>
      </li>
    </ul>
    <a
      data-testid="button-link"
      href="/find-business"
      onClick={[Function]}
    >
      <button
        className="btn btn--outline btn--medium"
        data-testid="button"
        type="button"
      >
        Find a Business
      </button>
    </a>
  </div>
</nav>
`;
